tests/main/test_generation_determinism.py

```
@@ -0,0 +1,115 @@
+            second.splitlines(),
+            fromfile="first",
+            tofile="second",
+            lineterm="",
+        )
+    )
+    pytest.fail(f"Generated output changed for {context}:\n{diff}", pytrace=False)
+
+
+def _generate_output_with_hash_seed(input_path: Path, input_file_type: InputFileType, seed: str) -> str:
+    env = {**os.environ, "PYTHONHASHSEED": seed}
+    result = subprocess.run(
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)